### DISCUSSION TITLE
Nuxt3 - Firebase: Error (auth/login-blocked), Production only!

### DISCUSSION THREAD

**Discussion by Kiansa** [vuejs/vuefire#1405] [Questions and Help] [answered]
On development everything works fine but when I build or generate, in both cases, on production I'm not able to authenticate and get this error:

`__session:1     Failed to load resource: the server responded with a status of 401 ()`
```
FirebaseError: Firebase: Error (auth/login-blocked).
    at Z2.runMiddleware (entry.9440d4e5.js:2443:472)
    at async eU._updateCurrentUser (entry.9440d4e5.js:2458:3303)
    at async mC (entry.9440d4e5.js:2758:116)
    at async pr.onAuthEvent (entry.9440d4e5.js:3013:704)
```
    

When user registers , I see that in firebase user has successfully added but on the client side throws the error above.
Same goes for sign in with (email/password. google or github).

Has anyone experienced error like this?

**Comment by johnpuaoi** [comment]
I just deployed a Nuxt 3 site last night to Netlify and have the same issue; when a user attempts to sign in, a "auth/login-blocked" error occurs. This only happens in production, built site locally and ran it in production mode to rule out anything to do with Netlify. 

This is the details shown with the error: "Error minting cookie: credential implementation provided to initializeapp() via the "credential" property failed to fetch a valid google oauth2 access token with the following error: "error fetching access token: error while making request: getaddrinfo enotfound metadata.google.internal. error code: enotfound".

Will be creating a barebones Nuxt project with Vuefire to see if this issue still occurs.

**Reply by johnpuaoi** [reply]
Just confirmed that the issue still persists in a new repo with just nuxt-vuefire and its dependencies. Also for additional info, I set the GOOGLE_APPLICATION_CREDENTIALS as an object. Again I can log in successfully with dev mode but auth fails with auth-blocked error in production after building.

**Comment by johnpuaoi** [comment]
UPDATE: I disabled SSR and redeployed my site and everything works as it should now. I guess this is a temp fix for now.

**Reply by Kiansa** [reply]
Thanks for the update, indeed disabling SSR fixes this issue. 
@posva do you mind looking into this? it's very strange, because setting ssr:false in routeRules for login and register pages do not fix the issue but if we put ssr:false for the whole app the problem goes away.

**Reply by johnpuaoi** [reply, marked as answer]
I opened an issue here: [#1408](https://github.com/vuejs/vuefire/issues/1408).

**Reply by dosstx** [reply]
Thanks for posting this. I too have the same issue: Works fine locally, but deploying to Netlify it fails with the same auth issue mentioned by you guys. Same thing happens if I deploy to Vercel with the exception that the error in browser is a little different:

Vercel:
```
entry.d6f1f58e.js:1  Uncaught (in promise) FetchError:  (404  (/api/__session)) //<-- Netlify says 502 instead of 404
    at async i (entry.d6f1f58e.js:1:93365)
    at async Nc (entry.d6f1f58e.js:3159:844)
```

It's possible that Firebase Auth is blocking because of error for security reasons.
